### PR TITLE
fix for  Flux plugin requires an additional config getter #3560 

### PR DESCRIPTION
**Pull request by gfichtenholt**
+ a few integration tests for flux v2 plug in were added/refactored

### CODE REVIEW

**Comment by gfichtenholt** [comment]
So there is quite a bit of "noise" in this PR, which for the most part feel free to ignore. The main part that's important is in utils.go, called newBackgroundClientGetter(). That's really the only important piece

File: cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/server.go [hunk 1]
```diff
@@ -71,7 +70,7 @@ func NewServer(configGetter server.KubernetesConfigGetter) (*Server, error) {
 		return nil, err
 	} else {
 		return &Server{
-			clientGetter:       clientGetter,
+			clientGetter:       newClientGetter(configGetter),
```
Comment (absoludity) [inline cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/server.go:73]: Great, so the cache is configured with the background client while the server (handling user requests) still always uses the provided config getter (that gets authz from the context).

File: cmd/kubeapps-apis/server/plugins.go [hunk 2]
```diff
@@ -352,11 +334,12 @@ func createConfigGetterWithParams(inClusterConfig *rest.Config, serveOpts ServeO
...
 func extractToken(ctx context.Context) (string, error) {
-	// per https://github.com/kubeapps/kubeapps/pull/3044
-	// extractToken() to return an empty token with a nil error if there is no metadata with the context.
+	// per https://github.com/kubeapps/kubeapps/issues/3560
+	// extractToken() to raise an error if there is no metadata with the context.
+	// note, the caller will wrap this as a codes.Unauthenticated status
 	md, ok := metadata.FromIncomingContext(ctx)
 	if !ok {
-		return "", nil
+		return "", fmt.Errorf("missing authorization metadata")
```
Comment (absoludity) [inline cmd/kubeapps-apis/server/plugins.go:342]: Excellent, thanks.

**Review by absoludity — APPROVED** [review]
Excellent, thanks Greg.

File: cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/server.go (hunk 1 above)
Comment (gfichtenholt) [reply cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/server.go:73]: correct

**Pull request merged**